### PR TITLE
 Add test functionality to core. 

### PR DESCRIPTION
**Pull request by nhaydel**
This PR adds the ability for rule authors to test a rule using the rule engine. The test function takes a json rule and the json datasets provided by the rule editor and runs the rule locally, producing the same excel output that the run validation function would produce. To run:

```
python core.py test -s sdtmig -v 3-4 -dp scripts/ae.json -r scripts/authored_whodrug_hierarchy_validation_rule.json --meddra ./meddra/ --whodrug ./whodrug/
```

### CODE REVIEW

**Comment by gcampion-cdisc** [comment]
Readme should be updated with description for validation options `-r, --rules TEXT`

**Comment by alexfurmenkov** [comment]
What is the difference between this command and `validate`?

**Comment by nhaydel** [comment]
> What is the difference between this command and `validate`?

This command is meant to allow users (mostly me trying to debug blocked rules) to run rules they have written in the rule editor that are not published. The only real difference between the two commands is that this one takes in a rule json file and a dataset json file and just validates that rule and dataset or set of datasets. I imagine rule authors can also use this to test engine functionality that is not yet in the editor like using external dictionaries

**Review by gcampion-cdisc — CHANGES_REQUESTED** [review]

File: core.py [hunk 1]
```diff
@@ -15,6 +15,9 @@
 from cdisc_rules_engine.enums.progress_parameter_options import ProgressParameterOptions
 from cdisc_rules_engine.enums.report_types import ReportTypes
 from cdisc_rules_engine.models.validation_args import Validation_args
+from cdisc_rules_engine.models.test_args import TestArgs
```
[the remark this answers is not in the document]
Comment (nhaydel) [reply core.py:18]: Added. If you are testing. Make sure that the datasets file looks like:
```
{
    "datasets": [{
      "filename": "cm.xpt",
      "label": "Concomitant/Concurrent medications",
      "domain": "CM",
      "variables": [...
```

[the remark this answers is not in the document]
Comment (gcampion-cdisc) [reply core.py:18]: i think this should also be mentioned in the readme

**Comment by nhaydel** [comment]
> > > What is the difference between this command and `validate`?
> > 
> > 
> > This command is meant to allow users (mostly me trying to debug blocked rules) to run rules they have written in the rule editor that are not published. The only real difference between the two commands is that this one takes in a rule json file and a dataset json file and just validates that rule and dataset or set of datasets. I imagine rule authors can also use this to test engine functionality that is not yet in the editor like using external dictionaries.
> > Note: This does come from a request from the rule authors to have a way to use the engine to test unpublished rules
> 
> Is there a reason why these can't use the same cli command?

Technically they could be, but I think it is better not to overload the validate command. The inputs are different, and I imagine the output may change in the future as well.

**Review by gcampion-cdisc — APPROVED** [review]
Sorry, my original review comment was delayed because I didn't realize it was still private (pending) until today.
I've made 2 more comments, but otherwise looks good and is working well.

**Pull request merged**